cmd/manager/main.go

```diff
@@ -107,12 +112,41 @@ func main() {
 		os.Exit(1)
 	}
 
-	// Create Service object to expose the metrics port.
-	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	// Add to the below struct any other metrics ports you want to expose.
+	servicePorts := []corev1.ServicePort{
+		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: corev1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: corev1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
+	}
+	// Create Service object to expose the metrics port(s).
+	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	if err != nil {
+		log.Info("Could not create metrics Service", "error", err.Error())
+	}
+
+	// Get the namespace the operator is currently deployed in.
+	depOperatorNs, err := k8sutil.GetOperatorNamespace()
 	if err != nil {
-		log.Info(err.Error())
+		log.Error(err, "Failed to get operator namespace")
+		os.Exit(1)
 	}
 
+	// CreateServiceMonitors will automatically create the prometheus-operator ServiceMonitor resources
+	// necessary to configure Prometheus to scrape metrics from this operator.
+	services := []*corev1.Service{service}
+	_, err = metrics.CreateServiceMonitors(cfg, depOperatorNs, services)
+	if err != nil {
+		log.Info("Could not create ServiceMonitor object", "error", err.Error())
+		// If this operator is deployed to a cluster without the prometheus-operator running, it will return
+		// ErrServiceMonitorNotPresent, which can be used to safely skip ServiceMonitor creation.
+		if err == metrics.ErrServiceMonitorNotPresent {
+			log.Info("Install prometheus-operator in your cluster to create ServiceMonitor objects", "error", err.Error())
+		}
+	}
+
+	if err = (&nodemaintenanceapi.NodeMaintenance{}).SetupWebhookWithManager(ctx, mgr); err != nil {
+		log.Error(err, "unable to create webhook", "webhook")
```

Comment:
Please have a look at the func signature.
`Error(err error, msg string, keysAndValues ...interface{})`
By just deleting "HyperConverged" there is an orphaned key now which won't be logged at all